### ISSUE TITLE
cargo flamegraph should set cargo environment variables

### ISSUE THREAD

**Issue by kuviman** [flamegraph-rs/flamegraph#244]
When using `cargo run` normally, cargo sets environment variables that can be detected from the binary

https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates

Would be nice if  running via `cargo flamegraph` would do same

**Comment by djc** [comment]
Sure, would you be able to submit a PR?

**Comment by kuviman** [comment]
I probably would be able, yes

I was wondering though if this is actually wanted. I dont know if people are using those variables from the binary and not during compilation, my use case might be too specific and i can easily work without this

**Comment by djc** [comment]
Yeah, I do think those variables are mainly intended for compilation time. Since flamegraph doesn't try to compile (it only reuses the Cargo metadata to find compilation targets IIRC), I don't think it would really apply here?
